packages/flutter_tools/lib/src/ios/mac.dart

```diff
@@ -281,6 +283,41 @@ Future<XcodeBuildResult> buildXcodeProject({
       ) ??
       <String, String>{};
 
+  final String? targetBuildDirPath = buildSettings['TARGET_BUILD_DIR'];
+  final Directory? targetBuildDir = targetBuildDirPath != null
+      ? globals.fs.directory(targetBuildDirPath)
+      : null;
+  final bool incrementalBuild = targetBuildDir != null && targetBuildDir.existsSync();
+
+  final buildCommands = <String>[
+    ...globals.xcode!.xcrunCommand(),
+    'xcodebuild',
+    '-configuration',
+    configuration,
+  ];
+
+  // Check the public headers before checking Xcode version so headers fingerprinter is created
+  // regardless of Xcode version.
+  final bool headersChanged = publicHeadersChanged(
+    environmentType: environmentType,
+    mode: buildInfo.mode,
+    buildDirectory: buildDirectoryPath,
+    artifacts: globals.artifacts,
+    fileSystem: globals.fs,
+    logger: globals.logger,
+  );
+  final Version? xcodeVersion = globals.xcode?.currentVersion;
+  if (headersChanged &&
+      incrementalBuild &&
+      (xcodeVersion != null && xcodeVersion >= Version(26, 0, 0))) {
+    // Xcode 26 changed the way headers are pre-compiled and will throw an error if the headers
+    // have changed since the last time they were compiled. To avoid this error, clean before
+    // building if headers have changed.
+    targetBuildDir.deleteSync(recursive: true);
```

Comment:
@bkonyi I added this to also delete the build directory before cleaning because I later discovered `xcodebuild` sometimes fails to delete it when cleaning

For reference and searchability, the error message given is:
```
Could not delete `/build/ios/Release-iphoneos` because it was not created by the build system and it is not a subfolder of derived data.
```